### ISSUE TITLE
Misaligned condition compilation and comments in switch statement

### ISSUE THREAD

**Issue by fphilipe** [nicklockwood/SwiftFormat#403]
Using the latest version 0.40.4 and the default options, the following code

```swift
func foo(n: Int) {
    switch n {
    // Comment
    case 1:
        break
    // Comment
    #if COMMENT_BEFORE
        case 2:
            break
    #endif
    // Comment
    case 3:
        break
    #if COMMENT_AFTER
        // Comment
        case 4:
            break
    #endif
    // Comment
    case 5:
        break
    // Comment
    #if COMMENT_BEFORE_AND_AFTER
        // Comment
        case 6:
            break
    #endif
    // Comment
    default:
        break
    }
}
```

gets formatted as this:

```swift
func foo(n: Int) {
    switch n {
    // Comment
    case 1:
        break
        // Comment
    #if COMMENT_BEFORE
        case 2:
            break
    #endif
    // Comment
    case 3:
        break
        #if COMMENT_AFTER
        // Comment
            case 4:
                break
    #endif
    // Comment
    case 5:
        break
        // Comment
        #if COMMENT_BEFORE_AND_AFTER
        // Comment
            case 6:
                break
    #endif
    // Comment
    default:
        break
    }
}
```

I'd expect the original code to remain unchanged.

What's wrong in my opinion:

- A comment before a conditional compilation condition gets indented.
- A comment after a conditional compilation condition causes the condition to get indented.

**Comment by nicklockwood** [comment]
@fphilipe thanks for reporting this. I think you're assessment of the issues is correct - I'll try to get this fixed asap.

**Comment by nicklockwood** [comment]
@fphilipe fixed in 0.40.5
